### ISSUE TITLE
KeyError: 'flags'

### ISSUE THREAD

**Issue by liliang-cn** [marella/ctransformers#70]
It's a M2 Macbook

```
from ctransformers import AutoModelForCausalLM

llm = AutoModelForCausalLM.from_pretrained(
    '/Users/liliang/Things/AI/lollms/config/models/c_transformers/llama-2-13b-chat.ggmlv3.q3_K_L.bin', model_type='llama')

print(llm('AI is going to'))
```

the result is as following:
```
Traceback (most recent call last):
  File "/Users/liliang/Things/AI/apps/roci/codes/service/app/app.py", line 3, in <module>
    llm = AutoModelForCausalLM.from_pretrained(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/liliang/anaconda3/envs/roci/lib/python3.11/site-packages/ctransformers/hub.py", line 157, in from_pretrained
    return LLM(
           ^^^^
  File "/Users/liliang/anaconda3/envs/roci/lib/python3.11/site-packages/ctransformers/llm.py", line 206, in __init__
    self._lib = load_library(lib, cuda=config.gpu_layers > 0)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/liliang/anaconda3/envs/roci/lib/python3.11/site-packages/ctransformers/llm.py", line 101, in load_library
    path = find_library(path, cuda=cuda)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/liliang/anaconda3/envs/roci/lib/python3.11/site-packages/ctransformers/lib.py", line 23, in find_library
    flags = get_cpu_info()["flags"]
            ~~~~~~~~~~~~~~^^^^^^^^^
KeyError: 'flags'
```

version 0.2.15 is fine, no errors.

**Comment by liliang-cn** [comment]
Seems like on Apple Silicon Macs `get_cpu_info()` doesn't return "flags"

**Comment by marella** [comment]
`get_cpu_info()` shouldn't be called for ARM processors which I will fix soon.

Can you please share the full output of `get_cpu_info()`. Just want to see if the "flags" info is present elsewhere.

```py
python3 -c 'import cpuinfo; print(cpuinfo.get_cpu_info())'
```

**Comment by marella** [comment]
This is fixed in the latest version 0.2.17 Please update and try again:

```sh
pip install 'ctransformers>=0.2.17'
```
